configure.ac

```diff
@@ -180,6 +180,24 @@ AS_CASE(["${build_os}"],
     # Clang on Solaris is largely untested.
     # https://bugs.ruby-lang.org/issues/17949
     AC_CHECK_TOOLS([CC], [cc gcc])
+],
+[darwin*], [
+    # /usr/bin/clang is always available as long as Command Line Tools or Xcode is installed.
+    # Prefer absolute path to avoid PATH lookup, which may lead to unaligned tool selection.
+    rb_darwin_default_CC=/usr/bin/clang
+    AC_MSG_CHECKING([for darwin default CC $rb_darwin_default_CC])
+    AS_IF([test -z "$CC" -a -x "$rb_darwin_default_CC"],
+        [
+            AC_MSG_RESULT([yes])
+            CC=$rb_darwin_default_CC
+        ],
+        [
+            AC_MSG_RESULT([no])
+            # Fall back to PATH lookup just in case the /usr/bin/clang is not
+            # available for some reason.
+            AC_CHECK_TOOLS([CC], [clang cc gcc c99])
+        ]
+    )
```

Comment:
What about expand `rb_CC` absolute path, only if the found `CC` is Xcode?

```
    AS_IF([$rb_CC -E -dM -xc - < /dev/null | grep -F __apple_build_version__ > /dev/null],
          [llvm_prefix= rb_CC=`command -v $rb_CC`], [llvm_prefix=llvm-])
```

Reply:
Not disagreeing with you as such, but it should be pointed out that at the least on Linux, the default assumption is that the entries in $PATH coming first, have higher precedence. So, for instance, if a user would have a PATH that first has /opt/bin/, and /usr/bin/ as second entry, then /opt/bin/ should be searched first. This will probably happen only very, very rarely, so it does not invalidate your statement, but I wanted to mention it in regards to (possible) user's expectations.

Reply:
I think we should abort in that case, if not `[test -x "${rb_prog}"]`, rather than searching from `$PATH`.

Reply:
It makes sense to me. It sounds well consistent behavior.

Reply:
I think `rb_darwin_default_CC` is not needed anymore.

Reply:
Sorry, do you mean we should find CC from PATH instead of having the default value?

Reply:
Yes, only for Xcode clang, as commented above.